kolibri/utils/server.py

```diff
@@ -115,8 +119,9 @@ def start_command(cls):
         thread.start()
 
     def run(self):
-        # Try to do the vacuum every 3 hours
-        call_command("vacuumsqlite", interval=10)
+        with lock:
+            # Try to do the vacuum every 3 hours
+            call_command("vacuumsqlite", interval=10)
```

Comment:
Isn't it the `vacuumsqlite` command that does the blocking wait? In that case, it should have the lock as an argument in the call or import it from `kolibri.utils.server`?

Reply:
...otherwise we risk a deadlock because it never yields the lock for the other waiting thread?